File: string/1 Check if string is palindrome/go/main.go
package main

import (
    "fmt"
)


func isPalindrome(str string) (bool) {

	l := 0
	r := len(str) - 1

	for r > l {
		if str[l] != str[r] {
			return false
		}
		l += 1
		r -= 1
	}

	return true
}

func main() {

	str := "asa"

	if isPalindrome(str) {
		fmt.Println("String is palindrome.")
	} else {
		fmt.Println("String is not palindrome.")
	}

}

File: string/1 Check if string is palindrome/python/main.py


def is_palindrome(str):

    l = 0
    r = len(str) - 1

    while (r > l):
        if (str[l] != str[r]):
            return False
        r -= 1
        l += 1

    return True

if __name__ == '__main__':

    str = 'sumeee'

    if is_palindrome(str):
        print('String is palindrome.')
    else:
        print('String is not palindrome.')